### DISCUSSION TITLE
how to maintain multiple versions of one project

### DISCUSSION THREAD

**Discussion by michaelbe812** [jscutlery/semver#707] [Q&A] [answered]
Hi,

I have following use case and would like to ask what would be the best way to do it using `jscutlery/semver`:

I have a project, lets just call it `devkit`. I want to maintain two versions of devkit:
- a 1.x.x for Angular Projects < 16.0.0
- a 2.x.x for Angular Projects >= 16.0.0
`devkit` should get bugfixes and features for both versions.

To achieve this I will need to have separate branches for both versions, let's say: `main` contains `devkit` 2.x.x and `devkit-1.x.x` is the branch for the 1.x.x version.

Is this the recommended approach? I remember that I faced some issues in the past when I did this (don't remember the errors though). However I just tried it in a fresh workspace and it seemed to work without any issues

**Comment by julianpoemp** [comment]
Did you find an answer to your question? I have a similar situation for my ngx-jodit project. I plan to use semver for different branches with two projects for each branch and different Angular 16 compatibility

**Reply by michaelbe812** [reply]
Hi @julianpoemp 
I just tried it myself. I created a repository with a main branch (holds current version, e.g 2.x.x) and one 1.x.x branch 

I did release multiple major minor and path versions switching the branches in between. 

I did not face any issue :) 

I hope this helps. If you want I can also share this repository with you. I should still have it (somewhere 😅) on my computer.

**Reply by julianpoemp** [reply]
that would be awesome!

**Reply by edbzn** [reply]
Hi, I also released a project from different branches and it works fine. I think that there's only one detail to mention, the `baseBranch` option should be updated according to the correct branch if you need to push, `main` is the default.

**Comment by michaelbe812** [comment, marked as answer]
> that would be awesome!

Here you go: https://github.com/mikelgo/semver-multi-branch-versioning 
Note: it is just a minimal version. I e.g. did not cherry pick commits from v1.x.x branch to the main branch (reflects the current version). You might want to this, e.g. when you need to bugfix something in multiple versions.

**Reply by julianpoemp** [reply]
thank you!

**Reply by michaelbe812** [reply]
You're welcome. I hope it helps
